aspnetcore/signalr/javascript-client.md

```diff
@@ -98,7 +98,140 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
+    try {
+        await connection.start();
+        console.assert(connection.state === signalR.HubConnectionState.Connected);
+        console.log("connected");
+    } catch (err) {
+        console.assert(connection.state === signalR.HubConnectionState.Disconnected);
+        console.log(err);
+        setTimeout(() => start(), 5000);
+    }
+};
+```
+
+If the client doesn't successfully reconnect within its first four attempts, the `HubConnection` will transition to the [Disconnected](javascript/api/%40aspnet/signalr/hubconnectionstate#reconnecting) state and fire its [onclosed](/javascript/api/%40aspnet/signalr/hubconnection#onreconnected) callbacks. This provides an opportunity to inform users the connection has been permanently lost and recommend refreshing the page:
+
+```javascript
+connection.onclose((error) => {
+  console.assert(connection.state === signalR.HubConnectionState.Disconnected);
+
+  document.getElementById("messageInput").disabled = true;
+
+  const li = document.createElement("li");
+  li.textContent = `Connection closed due to error "${error}". Try refreshing this page to restart the connection.`;
+  document.getElementById("messagesList").appendChild(li);
+})
+```
+
+In order to configure a custom number of reconnect attempts before disconnecting or change the reconnect timing, [withAutomaticReconnect](/javascript/api/%40aspnet/signalr/hubconnectionbuilder#withautomaticreconnect) accepts an array of numbers representing the delay in milliseconds to wait before starting each reconnect attempt.
+
+```javascript
+const connection = new signalR.HubConnectionBuilder()
+    .withUrl("/chatHub")
+    .withAutomaticReconnect([0, 0, 10000])
+    .build();
+
+    // .withAutomaticReconnect([0, 2000, 10000, 30000]) yields the default behavior
+```
+
+The preceding example configures the `HubConnection` to start attempting reconnects immediately after the connection is lost. This is also true for the default configuration.
+
+If the first reconnect attempt fails, the second reconnect attempt will also start immediately instead of waiting 2 seconds like it would in the default configuration.
+
+If the second reconnect attempt fails, the third reconnect attempt will start in 10 seconds which is again like the default configuration.
+
+The custom behavior then diverges again from the default behavior by stopping after the third reconnect attempt failure instead of trying one more reconnect attempt in another 30 seconds like it would in the default configuration.
+
+If you want even more control over the timing and number of automatic reconnect attempts, [withAutomaticReconnect](/javascript/api/%40aspnet/signalr/hubconnectionbuilder#withautomaticreconnect) accepts an object implementing the [IReconnectPolicy](/javascript/api/%40aspnet/signalr/ireconnectpolicy) interface, which has a single method named [nextRetryDelayInMilliseconds](/javascript/api/%40aspnet/signalr/ireconnectpolicy#nextRetryDelayInMilliseconds).
```

Comment:
@scottaddie Not all of these API doc pages exist yet. Considering that, is it still OK to squash and merge this PR? Or should I remove the dead links for now?

Reply:
@halter73 Thank you for pointing that out. I overlooked it. Please remove the dead links for now.

Reply:
@scottaddie I think bb4b4a4a226daf0f54701537353b6579d97983f1 removes all the dead links. Can you double check that everything is good to go now? Thanks.